### PR TITLE
Fix when selecting a saved payment method

### PR DESCRIPTION
**Pull request by porter-stripe**
## Summary
- When we would have something other than the saved payment method selected, then hit "View more" and select a payment method it would not be selected.
- The selected saved payment method should become the payment option to match the behavior of vertical mode and not require users to tap the saved payment method twice.
- I don't believe there to be any scenario where the saved payment method view controller completes with a selected payment method that we would not want it to be selected.

### Before
https://github.com/user-attachments/assets/31081f58-cedb-4003-a161-8bea4154b876

## Motivation
- Bug fix

## Testing
- Manual
- New UI test

## Changelog
N/A

### CODE REVIEW

File: Example/PaymentSheet Example/PaymentSheetUITest/EmbeddedUITest.swift [hunk 1]
```diff
@@ -389,6 +389,29 @@ class EmbeddedUITests: PaymentSheetUITestCase {
...
+        
+        // Select Cash App Pay
+        app.buttons["Cash App Pay"].waitForExistenceAndTap()
+        XCTAssertTrue(app.staticTexts["Cash App Pay"].waitForExistence(timeout: 10))
+        XCTAssertTrue(app.buttons["Checkout"].isEnabled)
+        
+        // Select a saved card
+        XCTAssertTrue(app.buttons["View more"].waitForExistenceAndTap())
+        XCTAssertTrue(app.buttons["•••• 4242"].waitForExistenceAndTap())
+
```
Comment (wooj-stripe) [inline Example/PaymentSheet Example/PaymentSheetUITest/EmbeddedUITest.swift:410]: Can we validate that the form has been dismissed somehow?

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift [hunk 2]
```diff
@@ -228,9 +228,11 @@ extension EmbeddedPaymentElement: VerticalSavedPaymentMethodsViewControllerDeleg
             savedPaymentMethods: latestPaymentMethods
         )
 
-        // If there are still saved payment methods & the saved payment method was previously selected to presenting
+        // Select the selected payment method if we selected a payment method and have more than one saved PM
+        // Or if there are still saved payment methods & the saved payment method was previously selected to presenting
         // the list of saved payment methods, then the embedded view should continue to show it is selected, otherwise unselected.
-        let isSelected: Bool = latestPaymentMethods.count > 0 && embeddedPaymentMethodsView.selection?.isSaved ?? false
+        let isSelected = (latestPaymentMethods.count > 1 && selectedPaymentMethod != nil) ||
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift:234]: is there any reason to check `latestPaymentMethods.count > 1` ? seems like we can just check `selectedPaymentMethod != nil || (embeddedPaymentMethodsView.selection?.isSaved ?? false && latestPaymentMethods.count > 0)`

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift [hunk 3]
```diff
@@ -228,9 +228,11 @@ extension EmbeddedPaymentElement: VerticalSavedPaymentMethodsViewControllerDeleg
             savedPaymentMethods: latestPaymentMethods
         )
 
-        // If there are still saved payment methods & the saved payment method was previously selected to presenting
+        // Select the selected payment method if we selected a payment method and have more than one saved PM
+        // Or if there are still saved payment methods & the saved payment method was previously selected to presenting
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift:232]: // Select the *saved payment method if we selected a payment method
// or
//  there are still saved payment methods & the saved payment method was previously selected to presenting

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift (hunk 2 above)
Comment (porter-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift:234]: Yes we do need this! The saved payment screen when you delete all but the last payment method auto selects it, even if the user did not tap it. I think i this case we should not auto select that saved payment method in the embedded view. I extended the existing test to cover this now.

Comment (wooj-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift:234]: Ah! right! thank you!

File: Example/PaymentSheet Example/PaymentSheetUITest/EmbeddedUITest.swift [hunk 4]
```diff
@@ -389,6 +389,49 @@ class EmbeddedUITests: PaymentSheetUITestCase {
...
+        XCTAssertTrue(app.staticTexts["Cash App Pay"].waitForExistence(timeout: 10))
+        XCTAssertTrue(app.buttons["Checkout"].isEnabled)
+        
+        // Select a saved card
+        XCTAssertTrue(app.buttons["View more"].waitForExistenceAndTap())
+        XCTAssertTrue(app.staticTexts["Select payment method"].waitForExistence(timeout: 10))
+        XCTAssertTrue(app.buttons["•••• 4242"].waitForExistenceAndTap())
+        
+        // Verify we have dismissed the saved payment method view and have the correct card selected
+        XCTAssertTrue(app.buttons["View more"].waitForExistence(timeout: 10))
```
[the remark this answers is not in the document]
Comment (porter-stripe) [reply Example/PaymentSheet Example/PaymentSheetUITest/EmbeddedUITest.swift]: yeah good idea

**Review by wooj-stripe — APPROVED** [review]

**Pull request merged**